clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -700,18 +700,37 @@ public FetchManagerMetrics(Metrics metrics, String metricGrpPrefix) {
         }
 
         public void recordTopicFetchMetrics(String topic, int bytes, int records) {
+            String topicMetricPrefix = "topic." + topic;
+
             // record bytes fetched
-            String name = "topic." + topic + ".bytes-fetched";
+            String name = topicMetricPrefix + ".bytes-fetched";
             Sensor bytesFetched = this.metrics.getSensor(name);
-            if (bytesFetched == null)
+            if (bytesFetched == null) {
                 bytesFetched = this.metrics.sensor(name);
+                bytesFetched.add(this.metrics.metricName(topicMetricPrefix + ".fetch-size-avg",
```

Comment:
Rather than prefixing each topic name with a topic prefix, I wonder if we should use a tag for the topic? This is how we handle node metrics in o.a.k.common.network.Selector.